### PR TITLE
Processing polygonize output fix

### PR DESCRIPTION
**Pull request by ghtmtt**
It should work now, but need some more test
I noticed that for GML it does not recognize the CRS of the input raster (tested on many different files). 
In fact 2 files are created: `file.gml` and `file.xsd` 
As suggested by @nyalldawson if I delete manually the `xsd` file and drag and drop the gml again in QGIS, then the correct `file.gfs` is created.

I don't understand if this a Processing issue when working with gml or something with my local configuration or something of gml file

Thanks for any feedback

### CODE REVIEW

**Comment by m-kuhn** [comment]
I don't think it's a local issue, I remember having similar issues. A little more information can be found in http://www.gdal.org/drv_gml.html . But I don't find any information about why an xsd is written.
Maybe @rouault can help with the logic behind it and say if it's better to store an xsd and/or gfs and how we can make processing create the correct file(s).

File: python/plugins/processing/tests/testdata/gdal_algorithm_tests.yaml [hunk 1]
```diff
@@ -177,3 +177,382 @@ tests:
...
+      COMPUTE_EDGES: true
+      INPUT:
+        name: dem.tif
+        type: raster
+    results:
+      OUTPUT:
+        hash: aba5167b8a2e97a93b22d406e0363f91c151fd353c0fc56ce24ac5fc
+        type: rasterhash
+
+  # - algorithm: gdalogr:polygonize
```
Comment (m-kuhn) [inline python/plugins/processing/tests/testdata/gdal_algorithm_tests.yaml]: Now that you fixed the issue, didn't you want to include this test again?

**Comment by rouault** [comment]
> if it's better to store an xsd and/or gfs

That depends. By default, the GML driver will write a .xsd as it is a standard mechanism. On opening, the GML driver will try to use the .xsd file if it exists, and if it can understand it (the driver can understand the .xsd it generates). If the .xsd doesn't exist or cannot be understood, then a first scan of the .gml file is done and a .gfs is generated. The slight advantage of the .gfs vs the .xsd is that the .gfs stores the feature count, spatial extent and SRS, which cannot be stored in a .xsd. So if you need to get those, the .gfs is more appropriate.

**Comment by m-kuhn** [comment]
Interestingly, the result is different if there is a GML opened with an `xsd` (ogr generated, as the result of a processing algorithm) and a GML opened with a `gfs` (ogr generated, as the result of a drag and drop operation). Does that sound plausible to you?
@ghtmtt the difference is in the detected crs, right?

**Comment by rouault** [comment]
Yes, in the .xsd case, the SRS isn't fetched. This could/should probably be fixed (but there's no such thing as a global SRS in a GML file. Features could potentially have different SRS, so you need to make some assumptions, like use the first SRS found and then potentially reproject further geometries)

**Comment by m-kuhn** [comment]
Sounds like a good approach from a QGIS view where one static [CS]RS per layer is expected.

For the processing unittest scenario here, it's probably the best approach to prefer gfs over xsd?
Can we somehow force ogr to already create a gfs from a processing algorithm by default?

**Comment by rouault** [comment]
> Can we somehow force ogr to already create a gfs from a processing algorithm by default?

It is a matter of deleting the .xsd and OGROpen()'ing the file.

**Comment by ghtmtt** [comment]
> It is a matter of deleting the .xsd and OGROpen()'ing the file. 

the option to re-opening the GML file after the creation is a good idea. IMHO, gfs is a little bit better than xsd.. maybe it is less confusing for the user

how can we process on that?

**Comment by ghtmtt** [comment]
@m-kuhn 

something is not working with the output file I guess. As said, xsd is created, I delete it, reload the gml so the gfs file is created and no more CRS warnings are shown. But I get this error:

```
FAIL: test_algorithms:29
5: 'Test (gdalogr:polygonize)', {'algorithm': 'gdalogr:polygonize', 'params': {'FI (__main__.TestGdalAlgorithms)
5: ----------------------------------------------------------------------
5: Traceback (most recent call last):
5:   File "/home/matteo/lavori/QGIS/QGIS/python/plugins/processing/tests/AlgorithmsTestBase.py", line 120, in check_algorithm
5:     self.check_results(alg.getOutputValuesAsDictionary(), defs['results'])
5:   File "/home/matteo/lavori/QGIS/QGIS/python/plugins/processing/tests/AlgorithmsTestBase.py", line 212, in check_results
5:     self.assertLayersEqual(expected_lyr, result_lyr, compare=compare)
5:   File "/home/matteo/lavori/QGIS/build-qgis3/output/python/qgis/testing/__init__.py", line 70, in assertLayersEqual
5:     _TestCase.assertEqual(self, layer_expected.dataProvider().crs().authid(), layer_result.dataProvider().crs().authid())
5: AssertionError: 'EPSG:4326' != ''
5: - EPSG:4326
```

It seems the CRS is not detected, right? Could it be a problem with the patch supplied?

**Comment by ghtmtt** [comment]
FYI:
Same test error for the contour algorithm. It didn't detected the CRS, xsd created, manually deleted, reload the gml file, gfs created (content seems corrected) but test fails (same error of above)

**Comment by m-kuhn** [comment]
@rouault do I understand the process correctly:

 * processing creates a new gml file with an xsd -> no CRS is assigned
 * first time opening in QGIS no gfs is found -> no CRS is assigned, gfs is created with the CRS found on the first feature
 * subsequent opening in QGIS gfs is found -> CRS is assigned

**Comment by ghtmtt** [comment]
@rouault @m-kuhn 
I just compiled QGIS, re-run the gdal polygonize tool and created the test with the GML output and all seems working know. 
Is something changed with the code?

If yes, could we drop this PR so I can restart a new and cleaner one with the added tests?

Thanks guys

**Comment by rouault** [comment]
@m-kuhn 

> do I understand the process correctly:
> 
>     processing creates a new gml file with an xsd -> no CRS is assigned
>     first time opening in QGIS no gfs is found -> no CRS is assigned, gfs is created with the CRS found on the first feature

Only if the .xsd has been deleted

>     subsequent opening in QGIS gfs is found -> CRS is assigned

Yes

**Comment by ghtmtt** [comment]
Now it seems that the .xsd is not created anymore. Directly .gml + .gfs created

**Comment by ghtmtt** [comment]
Nope, sorry.. just trash my last comment ;)

**Comment by nyalldawson** [comment]
What's the status here? is this just needing a rebase now?

**Comment by ghtmtt** [comment]
yep.. can close the PR, outdated. Thanks!

**Pull request closed without merging**